### PR TITLE
docs(examples): migrate custom data table state manager to vite

### LINKED ISSUE #15201: Migrate off CRA: packages/react/examples/custom-data-table-state-manager

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #15201

Migrate custom-data-table-state-manager example to vite

#### Changelog

**New**

- `packages/react/examples/custom-data-table-state-manager-vite`

**Removed**

- `packages/react/examples/custom-data-table-state-manager`

#### Testing / Reviewing

Check that example is working

### CODE REVIEW

File: packages/react/examples/custom-data-table-state-manager-vite/src/misc/enums.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const TABLE_SIZE = {
   /**
    * Regular size.
    */
-  REGULAR: 'normal',
+  REGULAR: 'lg',
```
Comment (guidari) [inline packages/react/examples/custom-data-table-state-manager-vite/src/misc/enums.js:18]: I switched the size name here to remove the warning. The size keeps the same on the app.

Screenshot 2023-11-29 at 10 41 32

**Review by guidari — APPROVED** [review]
LGTM 🚀

File: packages/react/examples/custom-data-table-state-manager-vite/src/misc/enums.js (hunk 1 above)
Comment (alisonjoseph) [reply packages/react/examples/custom-data-table-state-manager-vite/src/misc/enums.js:18]: Thanks!

**Comment by andreancardona** [comment]
@alisonjoseph @guidari had already approved I re-requested his review on accident. Going to bypass the requirement for now

**Review by andreancardona — APPROVED** [review]

**Pull request merged**